demisto_sdk/commands/generate_modeling_rules/generate_modeling_rules.py

```diff
@@ -0,0 +1,439 @@
+
+
+def replace_last_char(s: str) -> str:
+    """
+    Replaces the last char of the xif file to be ;
+    """
+    s = s[:-2]
+    s += ";\n"
+    return s
+
+
+def create_scheme_file(
+    mapping_list: List[MappingField], dataset_name, outputfile_schema
+):
+    """
+    Creates the .json schema file
+    """
+    logger.info("creating modeling rules schema\n")
+    name_type_dict = {}
+    for mapping_rule in mapping_list:
+        keys_list = mapping_rule.field_path_raw.split(".")
+        name = keys_list[0]
+        if name not in name_type_dict:
+            name_type_dict[name] = mapping_rule.create_schema_types()
+    modeling_rules_json = {dataset_name: name_type_dict}
+
+    with open(outputfile_schema, "w") as f:
+        res = json.dumps(modeling_rules_json, indent=4)
+        f.write(res)
+
+
+def process_yml_name(product: str, vendor: str):
+    """
+    Returns the name of the modeling rules capitalized
+    """
+    name = f"{product} {vendor} Modeling Rule\n"
+    name = name.replace("_", " ")
+    list_names = name.split()
+    capitalized_name_list = [name.capitalize() for name in list_names]
+    return " ".join(capitalized_name_list)
+
+
+def create_yml_file(outputfile_yml: Path, vendor: str, product: str, sdk_from_version):
+    """
+    Creates the yml file of the modeling rules
+    """
+    logger.info("creating modeing rules yml file\n")
+    max_version = get_max_version(
+        [
+            GENERAL_DEFAULT_FROMVERSION,
+            FILETYPE_TO_DEFAULT_FROMVERSION.get(FileType.MODELING_RULE, "6.10.0"),
+        ]
+    )
+    yml_file = (
+        f"fromversion: {max_version}\n"
+        f"id: {product}_{vendor}_modeling_rule\n"
+        f"name: {process_yml_name(product, vendor)}\n"
+        "rules: ''\n"
+        "schema: ''\n"
+        f"tags: {product}\n"
+    )
+
+    with open(outputfile_yml, "w") as f:
+        f.write(yml_file)
+
+
+def discoverType(value) -> str:
+    """
+    discovers the type of the event fiels and return a type compatible with the modeling rules schema
+    """
+    if isinstance(value, list):
+        return "array"
+    elif isinstance(value, bool):
+        return "bool"
+    elif isinstance(value, int):
+        return "int"
+    return "string"
+
+
+def extract_raw_type_data(event: dict, path_to_dict_field: str) -> tuple:
+    """
+    Extract the type of the field in the dict event
+    Args:
+        event (dict): A single raw event
+        path_to_dict_field (str): The path to the field in the raw event.
+    Returns:
+        (tuple): (type of the value - (str): is field of array type - (boolean))
+    """
+    if not event:
+        raise ValueError("The evnet provided is empty")
+    if not isinstance(event, dict):
+        raise ValueError("The array provided is not of type dict")
+
+    keys_split = path_to_dict_field.split(".")
+    temp: dict = event
+    for key in keys_split:
+        if isinstance(temp, dict):
+            temp = temp.get(key)  # type: ignore
```

Comment:
don't `type:ignore`. Please either check if the value `is None`, or, if absolutely necessary, use `type: ignore[specific code]`